packages/api/cms-api/src/user-permissions/user-permissions.service.ts

```diff
@@ -0,0 +1,167 @@
+import { EntityRepository } from "@mikro-orm/core";
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { Inject, Injectable } from "@nestjs/common";
+import getUuid from "uuid-by-string";
+
+import { CurrentUser } from "./dto/current-user";
+import { FindUsersArgs } from "./dto/paginated-user-list";
+import { User } from "./dto/user";
+import { UserContentScope, UserContentScopes } from "./entities/user-content-scopes.entity";
+import { UserPermission, UserPermissionSource } from "./entities/user-permission.entity";
+import { AvailableContentScopes } from "./user-content-scopes.resolver";
+import { UserPermissionConfigInterface, USERPERMISSIONS, USERPERMISSIONS_CONFIG_SERVICE } from "./user-permissions.types";
+import { getDate } from "./utils/getDate";
+@Injectable()
+export class UserPermissionsService {
+    constructor(
+        @Inject(USERPERMISSIONS_CONFIG_SERVICE) private readonly service: UserPermissionConfigInterface,
+        @InjectRepository(UserPermission) private readonly permissionRepository: EntityRepository<UserPermission>,
+        @InjectRepository(UserContentScopes) private readonly contentScopeRepository: EntityRepository<UserContentScopes>,
+    ) {}
+
+    async getAvailableContentScopes(): Promise<AvailableContentScopes[]> {
+        return this.service.getAvailableContentScopes
+            ? Object.entries(this.service.getAvailableContentScopes()).map(([scope, values]) => ({ scope, values }))
+            : [];
+    }
+
+    async getAvailablePermissions(): Promise<string[]> {
+        return [
+            "dam",
+            "pageTree",
+            "userPermissions",
```

Comment:
These should be constants in a `COMET_PERMISSIONS` array or similar

Reply:
I updated the types: https://github.com/vivid-planet/comet/pull/1223/commits/dc3ac32c3626c2465f42eab65d489068a6c73c98